data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -319,6 +323,23 @@ public boolean isReady() {
     return initialized;
   }
 
+  /**
+   * Determines the appropriate action for the given index, considering Data Stream requirements
+   */
+  private String determineAction(final String configuredAction, final String indexName) {
```

Comment:
I think a `DataStreamIndex` class would be useful here. It could provide the action override logic and the timestamp modifications. It would also be useful for any other checks we need.